### PR TITLE
Update to recipe for CMSSW 90X

### PR DESCRIPTION
**Pull request by edjtscott**
These two PRs have the necessary changes for the various rearrangements and improvements that Lindsey has made to the HGCal code, most of which is now included in CMSSW release 9_0_0_pre2. The main changes are:

-  Both the 2D clustering and multiclustering are now run by default in the reconstruction step, and the name of the module has become hgcalLayerClusters
-  The bug in the clustering associated with the use of pointers has been fixed
-  The HGCal has moved from D3 to D4 in CMSSW version 9
-  An option to do the multiclustering in real space (a bool called realSpaceCone) has been added

I've tested the full workflow (GSD, RECO, NTUP steps) for photons with pT=35 GeV and the usual validation plots are [here](http://escott.web.cern.ch/escott/HGCclustering/Pass6/Photon_Pt35/Unconverted/RebasedPre2fullTest/?match=SingleMC)

### CODE REVIEW

File: SubmitHGCalPGun.py [hunk 1]
```diff
@@ -146,6 +146,7 @@ def submitHGCalProduction():
             sys.exit()
     elif (opt.DTIER == 'RECO' or opt.DTIER == 'NTUP'):
         outDir=opt.inDir
+        #outDir='partGun_'+tag
```
Comment (clelange) [inline SubmitHGCalPGun.py]: You can probably leave out this file from the commits.

File: templates/partGun_GSD_template.py [hunk 2]
```diff
@@ -33,48 +33,12 @@
...
-                                                   simMuonDTDigis = cms.PSet(
-                initialSeed = cms.untracked.uint32(DUMMYSEED),
-                engineName = cms.untracked.string('TRandom3')
-                ),
-                                                   simMuonRPCDigis = cms.PSet(
-                initialSeed = cms.untracked.uint32(DUMMYSEED),
-                engineName = cms.untracked.string('TRandom3')
-                )
-                                                   )
+process.RandomNumberGeneratorService.generator.initialSeed = cms.untracked.uint32(DUMMYSEED)
```
Comment (clelange) [inline templates/partGun_GSD_template.py]: Did you check that if you run two jobs, the events are different by using only this line and removing those above?

File: templates/partGun_RECO_template.py [hunk 3]
```diff
@@ -107,28 +120,45 @@
...
...Halo2016Filter,process.Flag_globalSuperTightHalo2016Filter,process.Flag_HcalStripHaloFilter,process.Flag_hcalLaserEventFilter,process.Flag_EcalDeadCellTriggerPrimitiveFilter,process.Flag_EcalDeadCellBoundaryEnergyFilter,process.Flag_goodVertices,process.Flag_eeBadScFilter,process.Flag_ecalLaserCorrFilter,process.Flag_trkPOGFilters,process.Flag_chargedHadronTrackResolutionFilter,process.Flag_muonBadTrackFilter,process.Flag_trkPOG_manystripclus53X,process.Flag_trkPOG_toomanystripclus53X,process.Flag_trkPOG_logErrorTooManyClusters,process.Flag_METFilters,process.prevalidation_step,process.prevalidation_step1,process.prevalidation_step2,process.prevalidation_step3,process.prevalidation_step4,process.prevalidation_step5,process.prevalidation_step6,process.validation_step,process.validation_step1,process.validation_step2,process.validation_step3,process.validation_step4,process.validation_step5,process.validation_step6,process.dqmoffline_step,process.dqmoffline_1_step,process.dqmoffline_2_step,process.dqmoffline_3_step,process.dqmoffline_4_step,process.dqmofflineOnPAT_step,process.dqmofflineOnPAT_1_step,process.FEVTDEBUGHLToutput_step,process.MINIAODSIMoutput_step,process.DQMoutput_step)
```
Comment (clelange) [inline templates/partGun_RECO_template.py]: Are all these processes (in particular the DQM ones) really needed?

**Comment by edjtscott** [comment]
Have updated this now - DQM and validation gone, as has the submission file. Redid full tests, output is [here](http://escott.web.cern.ch/escott/HGCclustering/Pass6/Photon_Pt35/Unconverted/RebasedPre2AddressedTest/?match=SingleMC). Haven't added the random seed for the vertex smearing, I assume this can be done later?

**Comment by edjtscott** [comment]
@clelange the random seeds are back (haven't tested yet but should be fine)

**Comment by lgray** [comment]
Recall that this breaks further processing because it is an old list of random seeds and is not edited correctly by the "eras" process modifiers. The only random seeds you need to change is the one for the generator and the one for the vertex smearing. All the other seeds are randomized by having their inputs changed.

Suggest you revert back to what it was before.

**Comment by clelange** [comment]
So that means having the line
`process.RandomNumberGeneratorService.generator.initialSeed = cms.untracked.uint32(DUMMYSEED)`
back and also just adding a line:
`process.RandomNumberGeneratorService.VtxSmeared.initialSeed = cms.untracked.uint32(DUMMYSEED)`
That should do it, right?

**Comment by lgray** [comment]
Yep, that's all that's needed, and it is future proof against CMSSW changing the random seeds list underneath you.

**Comment by clelange** [comment]
OK, I just quickly did it inline to save Ed more hassle.

**Pull request merged**